src/Featurizers/Components/DocumentStatisticsEstimator.h

```diff
@@ -0,0 +1,334 @@
+///  \class         DocumentStatisticsAnnotationData
+///  \brief         Collection about statistics found within a number of
+///                 documents.
+///
+class DocumentStatisticsAnnotationData {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using FrequencyMap                      = std::unordered_map<std::string, std::uint32_t>;
+    using IndexMap                          = FrequencyMap;
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Data
+    // |
+    // ----------------------------------------------------------------------
+    FrequencyMap const                      TermFrequency;                  /// Words and the number of documents that it appears in
+    IndexMap const                          TermIndex;
+    std::uint32_t const                     TotalNumDocuments;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    DocumentStatisticsAnnotationData(FrequencyMap termFreq, IndexMap termIndex, std::uint32_t totalNumDocuments);
+    ~DocumentStatisticsAnnotationData(void) = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(DocumentStatisticsAnnotationData);
+};
+
+namespace Details {
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         DocumentStatisticsTrainingOnlyPolicy
+///  \brief         `DocumentStatisticsEstimator` implementation details.
+///
+class DocumentStatisticsTrainingOnlyPolicy {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using InputType                         = std::string;
+    using InputTypeConstIterator            = std::string::const_iterator;
+    using StringDecorator                   = std::function<std::string (InputTypeConstIterator begin, InputTypeConstIterator end)>;
+    using IndexMap                          = DocumentStatisticsAnnotationData::IndexMap;
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Data
+    // |
+    // ----------------------------------------------------------------------
+    static constexpr char const * const     NameValue = DocumentStatisticsEstimatorName;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    DocumentStatisticsTrainingOnlyPolicy(void);
+    DocumentStatisticsTrainingOnlyPolicy(StringDecorator decorator);
+
```

Comment:
Does this also need the `FEATURIZER_MOVE_CONSTRUCTOR_ONLY`?

Reply:
It seems like all TrainingOnlyPolicy class in Components miss it. Is it intentional? @davidbrownellWork

Reply:
I think it would better to refactor it in another PR if it's necessary. So I mark this conversation resolved at this time

Reply:
I agree with Ye. Technically speaking, the policy will never be copied. However, it is pretty subtle why it isn't required here but it is required just about everywhere else.